<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widgetry::DataBrowser</class>
 <widget class="QWidget" name="Widgetry::DataBrowser">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>733</width>
    <height>390</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Browser</string>
  </property>
  <layout class="QGridLayout" name="interfaceLayout" rowstretch="0,1,0" columnstretch="0,2">
   <item row="1" column="0" rowspan="2">
    <widget class="QFrame" name="filterFrame">
     <property name="frameShape">
      <enum>QFrame::Shape::StyledPanel</enum>
     </property>
     <layout class="QGridLayout" name="gridLayout_3" rowstretch="1,0,0">
      <item row="2" column="0">
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="2">
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="filterButton">
        <property name="text">
         <string>Filter</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="3">
       <widget class="QWidget" name="filterContainer" native="true">
        <layout class="QVBoxLayout" name="filterContainerLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
        </layout>
       </widget>
      </item>
      <item row="1" column="0" colspan="3">
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QWidget" name="topBar" native="true">
     <layout class="QHBoxLayout" name="topBarLayout" stretch="0,0,0,0">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="Widgetry::SearchBar" name="searchInput">
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toggleFiltersButtons">
        <property name="icon">
         <iconset resource="../resources/widgetry_resources.qrc">
          <normaloff>:/widgetry/icons/filter_icon.png</normaloff>:/widgetry/icons/filter_icon.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QWidget" name="buttonsContainer" native="true">
        <layout class="QHBoxLayout" name="buttonsContainerLayout">
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QToolButton" name="refreshButton">
           <property name="icon">
            <iconset resource="../resources/widgetry_resources.qrc">
             <normaloff>:/widgetry/icons/action_refresh.png</normaloff>:/widgetry/icons/action_refresh.png</iconset>
           </property>
           <property name="shortcut">
            <string>F5</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="addButton">
           <property name="icon">
            <iconset resource="../resources/widgetry_resources.qrc">
             <normaloff>:/widgetry/icons/action_add.png</normaloff>:/widgetry/icons/action_add.png</iconset>
           </property>
           <property name="shortcut">
            <string>Ctrl+N</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="editButton">
           <property name="icon">
            <iconset resource="../resources/widgetry_resources.qrc">
             <normaloff>:/widgetry/icons/action_edit.png</normaloff>:/widgetry/icons/action_edit.png</iconset>
           </property>
           <property name="shortcut">
            <string>Ctrl+E</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="deleteButton">
           <property name="icon">
            <iconset resource="../resources/widgetry_resources.qrc">
             <normaloff>:/widgetry/icons/action_delete.png</normaloff>:/widgetry/icons/action_delete.png</iconset>
           </property>
           <property name="shortcut">
            <string>Del</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1" rowspan="2">
    <widget class="Widgetry::TableWidget" name="tableWidget" native="true">
     <property name="minimumSize">
      <size>
       <width>400</width>
       <height>0</height>
      </size>
     </property>
    </widget>
   </item>
  </layout>
  <action name="actionShow">
   <property name="icon">
    <iconset resource="../resources/widgetry_resources.qrc">
     <normaloff>:/widgetry/icons/action_show.png</normaloff>:/widgetry/icons/action_show.png</iconset>
   </property>
   <property name="text">
    <string>Show</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Return</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionEdit">
   <property name="icon">
    <iconset resource="../resources/widgetry_resources.qrc">
     <normaloff>:/widgetry/icons/action_edit.png</normaloff>:/widgetry/icons/action_edit.png</iconset>
   </property>
   <property name="text">
    <string>Edit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionDelete">
   <property name="icon">
    <iconset resource="../resources/widgetry_resources.qrc">
     <normaloff>:/widgetry/icons/action_delete.png</normaloff>:/widgetry/icons/action_delete.png</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Widgetry::SearchBar</class>
   <extends>QLineEdit</extends>
   <header location="global">Widgetry/searchbar.h</header>
   <slots>
    <signal>searchRequested()</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>Widgetry::TableWidget</class>
   <extends>QWidget</extends>
   <header location="global">Widgetry/tablewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/widgetry_resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>toggleFiltersButtons</sender>
   <signal>toggled(bool)</signal>
   <receiver>filterFrame</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>313</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>searchInput</sender>
   <signal>searchRequested()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>refresh()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>280</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filterButton</sender>
   <signal>clicked()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>refresh()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow</sender>
   <signal>triggered()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>showCurrentItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionEdit</sender>
   <signal>triggered()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>editCurrentItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDelete</sender>
   <signal>triggered()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>deleteSelectedItems()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshButton</sender>
   <signal>clicked()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>refresh()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>621</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>addNewItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>651</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editButton</sender>
   <signal>clicked()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>editCurrentItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>681</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteButton</sender>
   <signal>clicked()</signal>
   <receiver>Widgetry::DataBrowser</receiver>
   <slot>deleteSelectedItems()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>711</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>366</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refresh()</slot>
  <slot>showCurrentItem()</slot>
  <slot>editCurrentItem()</slot>
  <slot>deleteSelectedItems()</slot>
  <slot>addNewItem()</slot>
 </slots>
</ui>
